From ed0218e5729eeb4bf0058fbdc3e8aa444109e6ca Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Fri, 10 Jul 2015 19:33:01 +0300
Subject: [PATCH] Make the ConsoleKit2 code optional

Add an ENABLE_CONSOLEKIT2 cmake option so that users and distros
can disable the ConsoleKit2 backend if it's not needed.
---
 CMakeLists.txt              | 6 ++++++
 src/daemon/PowerManager.cpp | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01ec7d6..7a53644 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
 # Options
 option(BUILD_MAN_PAGES "Build man pages" OFF)
 option(ENABLE_JOURNALD "Enable logging to journald" ON)
+option(ENABLE_CONSOLEKIT2 "Enable ConsoleKit2 support" ON)
 
 # Definitions
 add_definitions(-Wall -std=c++11)
@@ -141,6 +142,11 @@ else()
     set(REBOOT_COMMAND "/sbin/shutdown -r now")
 endif()
 
+if(ENABLE_CONSOLEKIT2)
+    add_definitions(-DHAVE_CONSOLEKIT2)
+    set(CMAKE_AUTOMOC_MOC_OPTIONS -DHAVE_CONSOLEKIT2)
+endif()
+
 # Set constants
 set(DATA_INSTALL_DIR            "${CMAKE_INSTALL_FULL_DATADIR}/sddm"                CACHE PATH      "System application data install directory")
 set(DBUS_CONFIG_DIR             "${CMAKE_INSTALL_SYSCONFDIR}/dbus-1/system.d"       CACHE PATH      "DBus config files directory")
diff --git a/src/daemon/PowerManager.cpp b/src/daemon/PowerManager.cpp
index 6329032..6be1cfb 100644
--- a/src/daemon/PowerManager.cpp
+++ b/src/daemon/PowerManager.cpp
@@ -117,9 +117,11 @@ namespace SDDM {
 #define LOGIN1_PATH     QStringLiteral("/org/freedesktop/login1")
 #define LOGIN1_OBJECT   QStringLiteral("org.freedesktop.login1.Manager")
 
+#ifdef HAVE_CONSOLEKIT2
 #define CK2_SERVICE  QStringLiteral("org.freedesktop.ConsoleKit")
 #define CK2_PATH     QStringLiteral("/org/freedesktop/ConsoleKit/Manager")
 #define CK2_OBJECT   QStringLiteral("org.freedesktop.ConsoleKit.Manager")
+#endif // HAVE_CONSOLEKIT2
 
     class SeatManagerBackend : public PowerManagerBackend {
     public:
@@ -200,9 +202,11 @@ namespace SDDM {
         if (interface->isServiceRegistered(LOGIN1_SERVICE))
             m_backends << new SeatManagerBackend(LOGIN1_SERVICE, LOGIN1_PATH, LOGIN1_OBJECT);
 
+#ifdef HAVE_CONSOLEKIT2
         // check if ConsoleKit2 interface exists
         if (interface->isServiceRegistered(CK2_SERVICE))
             m_backends << new SeatManagerBackend(CK2_SERVICE, CK2_PATH, CK2_OBJECT);
+#endif // HAVE_CONSOLEKIT2
 
         // check if upower interface exists
         if (interface->isServiceRegistered(UPOWER_SERVICE))
